pydatastructs/trees/tests/benchmarks/test_binary_trees.py

```diff
@@ -0,0 +1,32 @@
+import random, timeit, functools, os, pytest
+from pydatastructs.trees.binary_trees import (BinarySearchTree)
+from pydatastructs.utils.misc_util import Backend
+import random
+
+@pytest.mark.xfail
+def test_BST_insert(**kwargs):
+    cpp = Backend.CPP
+    repeat = 2
+    number = 2
+
+    size = int(os.environ.get("PYDATASTRUCTS_BENCHMARK_SIZE", "1000"))
+    size = kwargs.get("size", size)
+
+    def f(backend):
+        BST = BinarySearchTree
+        b = BST(backend=backend)
+        for node in range(-1000,1000):
+            b.insert(node, node)
+        for node in range(-1000, 1000):
+            b.search(node)
+        for node in range(2000):
+            b.delete(node)
```

Comment:
I see. Let's benchmark each operation separately? Like create 3 sub-functions, `f` (for `insert`), `g` (for `search`) and `h` (for `delete`). Pass the `b` to each of these.